cmsplugin_feed/processors.py

```diff
@@ -17,55 +17,68 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def remove_invalid(feed):
+    entries = feed.get('entries', [])
+    feed['entries'] = [e for e in entries if isinstance(e, dict)]
+    return feed
+
+
 def add_image_hrefs(feed):
     """ WARNING! it changes the feed arg """
     supported_image_types = ('image/jpeg', 'image/png')
-    entries = feed['entries']
+    entries = feed.get('entries', [])
     for entry in entries:
         if 'image' not in entry:
-            for link in entry.get('links', []):
-                if link.get('type') in supported_image_types:
-                    entry['image'] = link.get('href')
-                    break
+            try:
+                for link in entry.get('links', []):
+                    if link.get('type') in supported_image_types:
+                        entry['image'] = link.get('href')
+                        break
+            except (TypeError, AttributeError):
```

Comment:
Thanks.
The `TypeError` is only for something not iterable in `entry.get('links')`.